src/main/java/net/openhft/chronicle/queue/impl/ThreadGate.java

```diff
@@ -0,0 +1,9 @@
+package net.openhft.chronicle.queue.impl;
+
+/**
+ * A "gate" at which threads wait for some condition
+ */
+public interface ThreadGate {
+
+    void waitFor();
+}
```

Comment:
I looked but couldn't see a standard interface from `java.util.concurrency` to use here. If there was I'd be very open to using it. Also happy to bikeshed about the naming here, maybe "barrier" or something else is more appropriate?